src/client_side.cc

```diff
@@ -3953,9 +3953,15 @@ ConnStateData::terminateAll(const Error &rawError, const LogTagsErrors &lte)
     // may be set during the pipeline cleanup above
     connLeftovers_ = false;
 
-    if (intputToConsume && !inBuf.isEmpty()) {
-        debugs(83, 5, "forgetting client " << intputToConsume << " bytes: " << inBuf.length());
-        inBuf.clear();
+    if (!inBuf.isEmpty()) {
+        if (intputToConsume) {
+            debugs(83, 5, "forgetting client " << intputToConsume << " bytes: " << inBuf.length());
+            inBuf.clear();
+        } else {
+            bareError.update(ERR_STREAM_FAILURE);
```

Comment:
> Paraphrased PR description (to focus on one specific case): %err_code was empty for the third/unparsed pipeline transaction if something went wrong with the first transaction (e.g., client connection was aborted, server send an incomplete response body, etc.).

In this specific case, the second transaction will use some other %err_code, right? Why should not the third transaction use that other %err_code as well? In other words, why make that third transaction different from the second transaction as far as %err_code is concerned? Neither second nor third transaction are responsible for the error that killed the connection...

Reply:
I thought that the 'transaction-end-before-headers' transaction is a special one, and that is why we should mark it with a special error code. However I do not insist on that approach - we could use the existing rawError (if any) here and resort to ERR_STREAM_FAILURE if there is none.

Reply:
> I thought that the 'transaction-end-before-headers' transaction is a special one, and that is why we should mark it with a special error code.

It is certainly special in some aspects, but I do not see why it deserves a dedicated error code for this case. An access.log reader would still be able to distinguish transaction-end-before-headers transaction from other killed transactions on this connection AFAICT (by examining the logged request URI and connection ID). I do not see what _additional_ information a dedicated/new error code would provide in this case. Am I missing something?

> we could ... resort to ERR_STREAM_FAILURE if there is [no rawError].

Same logic should apply here: If there is no rawError, then it is a problem/bug, but that problem is not specific to the transaction-end-before-headers transaction. It affects _all_ current/pending transactions on this connection! I do not recall whether no-rawError scenario is possible, but if it is possible, it is a _different_ bug we ought to fix (in a dedicated PR that is not dedicated to transaction-end-before-headers transactions).

Reply:
> whether no-rawError scenario is possible

no-rawError case is when terminateAll() is called by ConnStateData::swanSong(). I assume that in this case Squid should have already responded with an error to existing transactions. However I do not see whether/where this assertion is verified.